src/limbo/mean/mean.hpp

```diff
@@ -0,0 +1,84 @@
+//|
+//| Contributor(s):
+//|   - Jean-Baptiste Mouret (jean-baptiste.mouret@inria.fr)
+//|   - Antoine Cully (antoinecully@gmail.com)
+//|   - Kontantinos Chatzilygeroudis (konstantinos.chatzilygeroudis@inria.fr)
+//|   - Federico Allocati (fede.allocati@gmail.com)
+//|   - Vaios Papaspyros (b.papaspyros@gmail.com)
+//|   - Roberto Rama (bertoski@gmail.com)
+//|
+//| This software is a computer library whose purpose is to optimize continuous,
+//| black-box functions. It mainly implements Gaussian processes and Bayesian
+//| optimization.
+//| Main repository: http://github.com/resibots/limbo
+//| Documentation: http://www.resibots.eu/limbo
+//|
+//| This software is governed by the CeCILL-C license under French law and
+//| abiding by the rules of distribution of free software.  You can  use,
+//| modify and/ or redistribute the software under the terms of the CeCILL-C
+//| license as circulated by CEA, CNRS and INRIA at the following URL
+//| "http://www.cecill.info".
+//|
+//| As a counterpart to the access to the source code and  rights to copy,
+//| modify and redistribute granted by the license, users are provided only
+//| with a limited warranty  and the software's author,  the holder of the
+//| economic rights,  and the successive licensors  have only  limited
+//| liability.
+//|
+//| In this respect, the user's attention is drawn to the risks associated
+//| with loading,  using,  modifying and/or developing or reproducing the
+//| software by the user in light of its specific status of free software,
+//| that may mean  that it is complicated to manipulate,  and  that  also
+//| therefore means  that it is reserved for developers  and  experienced
+//| professionals having in-depth computer knowledge. Users are therefore
+//| encouraged to load and test the software's suitability as regards their
+//| requirements in conditions enabling the security of their systems and/or
+//| data to be ensured and,  more generally, to use and operate it in the
+//| same conditions as regards security.
+//|
+//| The fact that you are presently reading this means that you have had
+//| knowledge of the CeCILL-C license and that you accept its terms.
+//|
+#ifndef LIMBO_MEAN_MEAN_HPP
+#define LIMBO_MEAN_MEAN_HPP
+
+#include <Eigen/Core>
+
+#include <limbo/tools/macros.hpp>
+
+namespace limbo {
+    namespace mean {
+        /** @ingroup mean
+          \rst
+          Base struct for mean definition.
+          \endrst
+        */
+        template <typename Params>
+        struct BaseMean {
+            BaseMean(size_t dim_out = 1) : _dim_out(dim_out) {}
+
+            template <typename GP>
+            Eigen::VectorXd operator()(const Eigen::VectorXd& v, const GP&) const
+            {
+                std::cerr << "'BaseMean' should never be called!" << std::endl;
+                assert(false);
+                return Eigen::VectorXd();
+            }
+
+            virtual size_t h_params_size() const { return 0; }
```

Comment:
Why do we have all these virtuals? there are useless here.

Reply:
> Why do we have all these virtuals? there are useless here.

True.. I still have OOP microbes in me..!